worlds/smw/Options.py

```diff
@@ -0,0 +1,214 @@
+import typing
+
+from Options import Choice, Range, Option, Toggle, DeathLink, DefaultOnToggle, OptionList
+
+
+class Goal(Choice):
+    """
+    Determines the goal of the seed
+    Bowser: Defeat Koopalings, reach Bowser's Castle and defeat Bowser
+    Yoshi Egg Hunt: Find a certain number of Yoshi Eggs
+    """
+    display_name = "Goal"
+    option_bowser = 0
+    option_yoshi_egg_hunt = 1
+    default = 0
+
+
+class BossesRequired(Range):
+    """
+    How many Bosses (Koopalings or Reznor) must be defeated in order to defeat Bowser
+    """
+    display_name = "Bosses Required"
+    range_start = 0
+    range_end = 11
+    default = 7
+
+
+class NumberOfYoshiEggs(Range):
+    """
+    How many Yoshi Eggs are in the pool for Yoshi Egg Hunt
+    """
+    display_name = "Total Number of Yoshi Eggs"
+    range_start = 1
+    range_end = 80
+    default = 50
+
+
+class PercentageOfYoshiEggs(Range):
+    """
+    What Percentage of Yoshi Eggs are required to finish Yoshi Egg Hunt
+    """
+    display_name = "Required Percentage of Yoshi Eggs"
+    range_start = 1
+    range_end = 100
+    default = 100
+
+
+class DragonCoinChecks(Toggle):
+    """
+    Whether collecting 5 Dragon Coins in each level will grant a check
+    """
+    display_name = "Dragon Coin Checks"
+
+
+class BowserCastleDoors(Choice):
+    """
+    How the doors of Bowser's Castle
+    Vanilla: Front and Back Doors behave as vanilla
+    Fast: Both doors behave as the Back Door
+    Slow: Both doors behave as the Front Door
+    """
+    display_name = "Bowser Castle Doors"
+    option_vanilla = 0
+    option_fast = 1
+    option_slow = 2
+    default = 0
+
+
+class LevelShuffle(Toggle):
+    """
+    Whether levels are shuffled
+    """
+    display_name = "Level Shuffle"
+
+
+class DisplaySentItemPopups(Choice):
+    """
+    What messages to display in-game for items sent
+    """
+    display_name = "Display Sent Item Popups"
+    option_none = 0
+    option_all = 1
+    default = 1
+
+
+class DisplayReceivedItemPopups(Choice):
+    """
+    What messages to display in-game for items received
+    """
+    display_name = "Display Received Item Popups"
+    option_none = 0
+    option_all = 1
+    option_progression = 2
+    default = 2
+
+
+class TrapFillPercentage(Range):
+    """
+    Replace a percentage of junk items in the item pool with random traps
+    """
+    display_name = "Trap Fill Percentage"
+    range_start = 0
+    range_end = 100
+    default = 0
```

Comment:
all option classes default to 0 fyi though this isn't problematic

Reply:
I prefer to be explicit.